### PR TITLE
Add tutorial how to program android apps

### PR DESCRIPTION
**Pull request by violinyanev**
Includes:

* Detailed step-by-step guide how to write a ramses app
* Formatted code snippets with explanations
* The source code of the "final" app

### CODE REVIEW

File: basics/android_app/app_src/app/src/main/java/com/example/myapplication/MonkeyThread.kt [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+        viewportWidth = camera?.getChild("viewport")?.getChild("width")
+        viewportHeight = camera?.getChild("viewport")?.getChild("height")
+        aspectRatio = camera?.getChild("frustum")?.getChild("aspectRatio")
+
+        lightId = getLogicNodeRootInput("LightControl")?.getChild("light_id")
+    }
+
+    override fun onUpdate() {
+        lightId?.set((frame / 60) % 3)
```
Comment (bojackHaasman) [inline basics/android_app/app_src/app/src/main/java/com/example/myapplication/MonkeyThread.kt:30]: this will overflow at some point right?

**Review by bojackHaasman — CHANGES_REQUESTED** [review]
In general very cool tutorial.
one remark: maybe state in which file you are operating everytime before you explain code

File: basics/android_app/app_src/app/build.gradle [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+dependencies {
+
+    implementation 'androidx.core:core-ktx:1.7.0'
+    implementation 'androidx.appcompat:appcompat:1.4.2'
+    implementation 'com.google.android.material:material:1.6.1'
+    implementation 'androidx.constraintlayout:constraintlayout:2.1.4'
+    testImplementation 'junit:junit:4.13.2'
+    androidTestImplementation 'androidx.test.ext:junit:1.1.3'
+    androidTestImplementation 'androidx.test.espresso:espresso-core:3.4.0'
+    implementation "io.github.bmwcarit:ramses-aar:1.0.3"
```
Comment (bojackHaasman) [inline basics/android_app/app_src/app/build.gradle:47]: go over dependencies again and verify which ones you really need

File: basics/android_app/app_src/app/src/main/java/com/example/myapplication/MainActivity.kt [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        MonkeyThread("MonkeyThread", applicationContext)
+    }
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        // Loads the scene from its binary asset files
+        monkeyThread.initRamsesThreadAndLoadScene(assets, "monkey.ramses", "monkey.rlogic")
+
+        val surfaceView = findViewById<SurfaceView>(R.id.surfaceView)
```
Comment (bojackHaasman) [inline basics/android_app/app_src/app/src/main/java/com/example/myapplication/MainActivity.kt:30]: pls add comment that ramses renders to the surface of the surfaceview

File: basics/android_app/app_src/build.gradle [hunk 4]
```diff
@@ -0,0 +1,10 @@
+// Top-level build file where you can add configuration options common to all sub-projects/modules.
+plugins {
+    id 'com.android.application' version '7.2.0' apply false
+    id 'com.android.library' version '7.2.0' apply false
```
Comment (bojackHaasman) [inline basics/android_app/app_src/build.gradle:4]: do you need this? we don't have a library module in the ramses-composer repo, right?

File: basics/android_app/app_src/settings.gradle [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+        mavenCentral()
+    }
+}
+dependencyResolutionManagement {
+    repositoriesMode.set(RepositoriesMode.FAIL_ON_PROJECT_REPOS)
+    repositories {
+        google()
+        mavenCentral()
+    }
+}
```
Comment (bojackHaasman) [inline basics/android_app/app_src/settings.gradle:14]: do you know how this works, I have never seen this part

**Review by bojackHaasman — CHANGES_REQUESTED** [review]

**Review by ParadoxDaKo — APPROVED** [review]
Seems OK (from Ramses Composer side)

**Pull request merged**